### ISSUE TITLE
Provide arguments when launching the interpreter from the UI

### ISSUE THREAD

**Issue by echebbi** [gemoc/ale-lang#118]
## Motivation

Executions sometimes require inputs from the user:

```kotlin
open class FSM {
	
    @main
    def void run(String event) {
        self.initialState.on(event);
    }
	
}
```

However it is currently impossible to pass arguments when launching the interpreter from the IDE.

## Desired behavior

The *ALE Application* launch configuration allows to set arguments.

## Which kind of arguments are allowed?

It would be simple to provide a text field that can be filled with comma-separated arguments but that would just allow parseable values (string, numbers, booleans).

We may want to let the user specify more complex data and thus handle Ecore models as well.

## Implementation

1. Enhance the [AleLaunchConfigurationTab](https://github.com/gemoc/ale-lang/blob/15027a3aeaaf8e00ff5e16488575c14d3a7b74e1/plugins/org.eclipse.emf.ecoretools.ale.ide.ui/src/org/eclipse/emf/ecoretools/ale/ide/ui/launchconfig/AleLaunchConfigurationTab.java) to handle arguments
2. Update the [AleEvaluationJob](https://github.com/gemoc/ale-lang/blob/15027a3aeaaf8e00ff5e16488575c14d3a7b74e1/plugins/org.eclipse.emf.ecoretools.ale.ide.ui/src/org/eclipse/emf/ecoretools/ale/ide/ui/launchconfig/AleEvaluationJob.java#L143) to pass these arguments to the interpreter

**Comment by dvojtise** [comment]
The strategy using GEMOC is not to add the parameter on the main method but on the `@init` method. This allows separating the role of initializing the runtime data and the start.

please have a look to the official example available in GEMOC 
![image](https://user-images.githubusercontent.com/661468/76848918-c2979400-6844-11ea-9983-e564461fbe2f.png)

![image](https://user-images.githubusercontent.com/661468/76849188-389bfb00-6845-11ea-9486-8b677227f135.png)



So I suggest that if the launch conf is improved to handle arguments, they should reuse the existing approach and `@init` annotation.

**Comment by echebbi** [comment]
Makes sense. Moreover providing an `@init` annotation may be a way to support #111 that asks for a way to initialize objects.

For the record, the article [Why Kermeta doesn't have constructors?](http://www.kermeta.org/documents/faq/constructors.html) may be worth reading.
